src/displayapp/screens/settings/SettingSetTime.cpp

```diff
@@ -105,38 +121,90 @@ SettingSetTime::SettingSetTime(Pinetime::Applications::DisplayApp *app, Pinetime
   lv_obj_align(btnSetTime, lv_scr_act(), LV_ALIGN_IN_BOTTOM_MID, 0, 0);
   lv_obj_set_style_local_value_str(btnSetTime, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, "Set");
   lv_obj_set_event_cb(btnSetTime, event_handler);
+
+  if (settingsController.GetClockType() == Controllers::Settings::ClockType::H12) {
+    struct Time12H convertedTime = timeConvert(hoursValue);
+    if (hoursValue < 12) {
+      lv_label_set_text_static(lblampm, "AM");
+    } else {
+      lv_label_set_text_static(lblampm, "PM");
+    }
+    lv_label_set_text_fmt(lblHours, "%02d", convertedTime.hours);
+  }
 }
 
 SettingSetTime::~SettingSetTime() {
   lv_obj_clean(lv_scr_act());
 }
 
-void SettingSetTime::HandleButtonPress(lv_obj_t *object, lv_event_t event) {
+Time12H timeConvert(int time24H) {
+  struct Time12H time12H;
+  switch (time24H) {
+    case 0:
+      time12H.hours = 12;
+      time12H.ampm = "AM";
+      break;
+    case 1 ... 11:
+      time12H.hours = time24H;
+      time12H.ampm = "AM";
+      break;
+    case 12:
+      time12H.hours = 12;
+      time12H.ampm = "PM";
+      break;
+    case 13 ... 23:
+      time12H.hours = time24H - 12;
+      time12H.ampm = "PM";
+      break;
+  }
+  return time12H;
+}
+
+void SettingSetTime::HandleButtonPress(lv_obj_t* object, lv_event_t event) {
+  bool is24H;
   if (event != LV_EVENT_CLICKED)
     return;
 
+  is24H = (settingsController.GetClockType() == Controllers::Settings::ClockType::H24);
```

Comment:
This variable is still redundant. Do you have it so the code looks neater?

Reply:
Would it be better practice to check `settingsController.GetClockType()` in place of is24H? Or is there a variable that contains this info already that I am missing?

Reply:
I would prefer that for now.

Reply:
I see. I have made that change for the next commit.